.github/workflows/dart.yml

```diff
@@ -9,42 +9,59 @@ on:
   - cron: "0 0 * * 0" # Run every Sunday at 00:00.
 
 jobs:
+
+  # Default test configurations.
   test:
     runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: example
     strategy:
-      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        # Test latest stable, beta, dev channels + last stable release
-        # prior to the introduction of the unified `dart` developer tool.
-        sdk: [stable, beta, dev, 2.9.3, 2.12.0-29.10.beta]
```

Comment:
The specific beta case was testing this branch:
https://github.com/dart-lang/setup-dart/blob/main/setup.sh#L27

If we have no case were we test "detect channel from version anymore" we lose coverage.

Reply:
Gotcha - I'll update this PR to test inferring the channel from the sdk param.